pkg/didcomm/protocol/exchange/exchange.go

```diff
@@ -86,29 +91,39 @@ func GenerateInviteWithKeyAndEndpoint(invite *Invitation) (string, error) {
 	return encodedExchangeInvitation(invite)
 }
 
+// Client for exchange protocol
+type Client struct {
+	prov transport.OutboundTransport
+}
+
+// New instanstiated new exchange client
+func New(prov Provider) *Client {
+	return &Client{prov: prov}
+}
+
 // SendExchangeRequest sends exchange request
-func SendExchangeRequest(exchangeRequest *Request, destination string, transport transport.OutboundTransport) error {
+func (c *Client) SendExchangeRequest(exchangeRequest *Request, destination string) error {
```

Comment:
As a followup, we should see if we should be using request contexts for Cancellables (I haven't thought too much about the implications of simply using the http client timeout duration yet).

Reply:
can we do this in separate issue/pr ?

Reply:
Yes. Created https://github.com/hyperledger/aries-framework-go/issues/109